dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/GlobalTaskDispatchWaitingQueueLooper.java

```diff
@@ -65,24 +66,20 @@ public void run() {
     }
 
     void doDispatch() {
-        final ITaskExecutionRunnable taskExecutionRunnable = globalTaskDispatchWaitingQueue.takeTaskExecuteRunnable();
-        final TaskInstance taskInstance = taskExecutionRunnable.getTaskInstance();
-        try {
-            final TaskExecutionStatus status = taskInstance.getState();
-            if (status != TaskExecutionStatus.SUBMITTED_SUCCESS && status != TaskExecutionStatus.DELAY_EXECUTION) {
-                log.warn("The TaskInstance {} state is : {}, will not dispatch", taskInstance.getName(), status);
-                return;
-            }
-            taskExecutorClient.dispatch(taskExecutionRunnable);
-        } catch (Exception e) {
+        ITaskExecutionRunnable taskExecutionRunnable =
+                globalTaskDispatchWaitingQueue.takeTaskExecuteRunnable();
+        boolean addTaskSuccess = workerGroupTaskDispatcherManager.addTaskToWorkerGroup(
+                taskExecutionRunnable.getTaskInstance().getWorkerGroup(),
+                taskExecutionRunnable, 0);
+        if (!addTaskSuccess) {
+            log.warn("worker group is deleting or deleted, taskInstance: {}", taskExecutionRunnable.getTaskInstance());
```

Comment:
```suggestion
            log.warn("Dispatch TaskInstance: {} WorkerGrouTaskDispatcher: {} failed", taskExecutionRunnable.getTaskInstance().getName(), taskExecutionRunnable.getTaskInstance().getWorkerGroup());
```

Reply:
Thank you very much for your suggestions! I completely agree with your changes and they do make the code clearer. I will make the changes according to your suggestions.